File: backend/SistemaComercial.Domain/Entities/Configuracion.cs


public class Configuracion
{
    public short IdConfiguracion { get; set; }

    public string NombreNegocio { get; set; } = string.Empty;

    public string? RazonSocial { get; set; }

    public string? Cuit { get; set; }
    public string? Direccion { get; set; }

    public string? Telefono { get; set; }

    public string? Email { get; set; }
    public string? Logo { get; set; }

    public string Moneda { get; set; } = string.Empty;

    public bool PermitirStockNegativo { get; set; } = false;

    public bool SugerirPrecioVenta { get; set; } = true;
    public int StockMinimoDefecto { get; set; } = 0;

    public short IntentosLogin { get; set; } = 5;
}

File: frontend/src/api/configuracion.ts
import { api } from "./axios";

export interface Configuracion {

    idConfiguracion: number;

    nombreNegocio: string;

    razonSocial: string;

    cuit: string;

    direccion: string;

    telefono: string;

    email: string;

    logo: string;

    moneda: string;

    permitirStockNegativo: boolean;

    sugerirPrecioVenta: boolean;

    stockMinimoDefecto: number;

    intentosLogin: number;

}

export async function obtenerConfiguracion() {

    const response = await api.get<Configuracion>(
        "/Configuracion",
    );

    return response.data;

}